gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -86,8 +86,10 @@ private boolean includeField(Field f, boolean serialize) {
   private List<String> getFieldNames(Field f) {
     SerializedName annotation = f.getAnnotation(SerializedName.class);
     if (annotation == null) {
-      String name = fieldNamingPolicy.translateName(f);
-      return Collections.singletonList(name);
+      String fieldName = fieldNamingPolicy.translateName(f);
+      List<String> fieldNames = new ArrayList<>(fieldNamingPolicy.translateToAlternateNames(f));
+      fieldNames.add(fieldName);
```

Comment:
For consistency with `@SerializedName` should probably first add `fieldName` (so it will be used for serialization and deserialization by default), and then the 'alternate names' so that they are only used as fallback for deserialization.

And then maybe also add a test which verifies that when 'alternate names' are configured they don't affect serialization, and only affect deserialization.